src/devices/bus/mc10/mc10cart.cpp

```diff
@@ -0,0 +1,239 @@
+// license:BSD-3-Clause
+// copyright-holders:tim lindner
+/*********************************************************************
+
+    mc10cart.cpp
+
+    MC-10 / Alice expansion slot - typically used for the 16K external
+    RAM expansion.
+
+         MC-10 and Alice pinout listing
+       ---  -------           ---  -------
+         1  GND                18  A6
+         2  GND                19  A7
+         3  D0                 20  A8
+         4  D1                 21  A9
+         5  D2                 22  A10
+         6  D3                 23  A11
+         7  D4                 24  A12
+         8  D5                 25  A13
+         9  D6                 26  A14
+        10  D7                 27  A15
+        11  R/!W               28  E
+        12  A0                 29  SEL
+        13  A1                 30  RESET
+        14  A2                 31  !NMI
+        15  A3                 32  +5V
+        16  A3                 33  GND
+        17  A5                 34  GND
+
+	SEL is an input to the MC-10 that allows the cartridge to remove
+	the internal chips from the bus.
```

Comment:
Can hardware in the cartridge control `SEL` on-the-fly?  If it can, the host should probably install the chips affected by it in a view so it can unmap them when necessary.

Reply:
Yes. I see the logic in your suggestion, but that adds a view that has to be explicitly manager by trampolines. By having the slot device create the view, it can control access to the internal bus in exactly the same way the hardware does.